### DISCUSSION TITLE
Appending model attributes that use Attribute casts is broken in v2.11.1

### DISCUSSION THREAD

**Discussion by realslimsutton** [livewire/livewire#5535] [7. Bugs (archived)] [answered]
This was broken in #5525.

Attributes can be created in a model using the [Laravel Attribute cast](https://laravel.com/docs/9.x/eloquent-mutators#defining-an-accessor). These attributes can then be appended to the json object of a model using the `appends` property within the model.
[This behaviour is documentated in the latest Laravel docs (and was originally introduced in Laravel 8)](https://laravel.com/docs/9.x/eloquent-serialization#appending-values-to-json).

In the scenario we have a model called `Author` with database column of `name`. This model may have two attributes could be defined, and appended to the model json like described above:

```
protected $appends = [
     'first_name',
    'last_name'
];

public function firstName(): Attribute
{
    return Attribute::make(
        function() {
            return Str::before($this->name, ' ');
        }
    );
}

public function lastName(): Attribute
{
    return Attribute::make(
        function() {
            return Str::after($this->name, ' ');
        }
    );
}
```

Since now that snake case is disabled, when Livewire attempts to serialize the model during hydration, a `BadMethodCallException` is thrown as Laravel falls back on the older method of defining an attribute. Laravel would be expecting a method such as `getFirstNameAttribute()` to be defined whereas we've defined `firstName`.

**Comment by realslimsutton** [comment]
Anyone else experiencing this issue can for now downgrade back to Livewire v2.11.0 (though be aware you won't have the latest fixes):
`composer require livewire/livewire:2.11.0`

**Comment by joshhanley** [comment]
For anyone else that has comes across this, @realslimsutton has submitted a PR #5536 with a fix. 🙂

**Comment by joshhanley** [comment, marked as answer]
We've decided to revert the Livewire PR that caused this issue and have tagged a new release of Livewire v2.11.2. See my comment on PR https://github.com/livewire/livewire/pull/5536#issuecomment-1407813069 for more details on why it was reverted.
